rotkehlchen/chain/bitcoin_cash/__init__.py

```diff
@@ -0,0 +1,47 @@
+from typing import Dict, List
+
+import requests
+
+from rotkehlchen.chain.bitcoin_cash.utils import cash_to_legacy_address
+from rotkehlchen.errors.misc import RemoteError, UnableToDecryptRemoteData
+from rotkehlchen.fval import FVal
+from rotkehlchen.types import BTCAddress
+from rotkehlchen.utils.misc import satoshis_to_btc
+from rotkehlchen.utils.network import request_get
+
+
+def get_bitcoin_cash_addresses_balances(accounts: List[BTCAddress]) -> Dict[BTCAddress, FVal]:
+    """Queries api.haskoin.com for the balances of BCH accounts.
+
+    May raise:
+    - RemoteError if there is a problem querying api.haskoin.com
+    """
+    source = 'https://api.haskoin.com'
+    balances: Dict[BTCAddress, FVal] = {}
+    try:
+        accounts_chunks = [accounts[x:x + 80] for x in range(0, len(accounts), 80)]
+        for accounts_chunk in accounts_chunks:
+            params = ','.join(accounts_chunk)
+            bch_resp = request_get(url=f'{source}/bch/address/balances?addresses={params}')
+            for entry in bch_resp:
+                # Try and get the initial address passed to the request.
+                # This is because the API only returns CashAddr format as response.
+                if entry['address'] in accounts_chunk:
```

Comment:
In this loop you will need to:

use `deserialize_fval` for any FVals, and catch `DeserializationError` if something is wrong. Otherwise with the code as is right now if say `entry['confirmed']` is not really a number it will go BOOM.